x-pack/filebeat/input/awss3/s3.go

```diff
@@ -110,3 +128,29 @@ func getProviderFromDomain(endpoint string, ProviderOverride string) string {
 	}
 	return "unknown"
 }
+
+func configureAccessPointEndpoint(awsConfig *awssdk.Config, accessPointARN string) error {
+	// When using the access point ARN, requests must be directed to the
+	// access point hostname. The access point hostname takes the form
+	// AccessPointName-AccountId.s3-accesspoint.Region.amazonaws.com
+	parts := strings.Split(accessPointARN, ":")
+	region := parts[3]
+	accountID := parts[4]
+	accessPointName := strings.Split(parts[5], "/")[1]
+
+	// Construct the endpoint for the Access Point
+	endpoint := fmt.Sprintf("%s-%s.s3-accesspoint.%s.amazonaws.com", accessPointName, accountID, region)
+
+	// Set up a custom endpoint resolver for Access Points
+	//nolint:staticcheck // haven't migrated to the new interface yet
+	awsConfig.EndpointResolverWithOptions = awssdk.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (awssdk.Endpoint, error) {
```

Comment:
I would expect that the SDK would handle taking the ARN -- is there documentation that outlines that we have to manually string together this value and make it into an endpoint resolver?

Reply:
> is there documentation that outlines that we have to manually string together this value and make it into an endpoint resolver?

I followed these guidelines from this documentation (https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html) where it says:
```
Access points - When you use this action with an access point, you must provide the alias of the access point in place of the bucket name or specify the access point ARN.
When using the access point ARN, you must direct requests to the access point hostname.
The access point hostname takes the form AccessPointName-AccountId.s3-accesspoint.Region.amazonaws.com.
When using this action with an access point through the AWS SDKs, you provide the access point ARN in place of the bucket name. 
For more information about access point ARNs, see [Using access points](https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-access-points.html) in the Amazon S3 User Guide.
```

I understood there that redirecting the requests to the custom endpoint was our responsibility, but you are right, I just tried letting that to the SDK and it works the same. I'll update the PR to remove that manual assignment.

> Is this EndpointResolver is also shared with the SQS connection? Does SQS work when using an access point ARN?

SQS doesn't work with the Access Point ARN in the same way it doesn't work with the bucket ARN currently. From https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-aws-s3.html:
```
SQS notification method is enabled setting queue_url configuration value.
S3 bucket list polling method is enabled setting bucket_arn configuration value.
Both value cannot be set at the same time, at least one of the two value must be set.
```

The input takes different paths at the beginning based on whether `queue_url` is configured for SQS, or `bucket_arn`: https://github.com/elastic/beats/blob/main/x-pack/filebeat/input/awss3/input.go#L55-L61. Now, `access_point_arn` would be another alternative for using the S3 list polling so it'd take the second path.

Reply:
https://github.com/elastic/beats/pull/41495/commits/b4a41e4cbbd133c4032a75972b2f1e5ca164f557

Reply:
Thank you!